### PR TITLE
#23 hyper cube layout with multiple q data pages

### LINKED ISSUE #23: Add support for HyperCubeLayout with multiple qDataPages.
Add support for script generation from HyperCubeLayout with multiple qDataPages.

### PR DESCRIPTION
**Pull request by KFredberg**
Fixes #23.

### Status
```
[ ] Under development
[X] Waiting for code review
[ ] Waiting for merge
```
### Information
```
[ ] Contains breaking changes
[ ] Contains new API(s)
[ ] Contains documentation
[X] Contains test
```

### CODE REVIEW

**Comment by KFredberg** [comment]
I am not able to run coverage task. Is it because .coveralls.yml is missing?

**Comment by KFredberg** [comment]
I've got it working by signing up to coveralls.io and link the repo. After that I needed to add an enviroment variable COVERALLS_REPO_TOKEN to circleci with the coveralls repo_token. It would be good if the contribution.md would include guides to how to setup circleci and coveralls.

File: src/hyper-cube.js [hunk 1]
```diff
@@ -51,22 +51,57 @@ class HyperCube {
...
+      if (dataPage.qArea && dataPage.qArea.qTop < qHeight) {
+        throw new Error('qDataPages have overlapping data pages.');
+      }
 
-    throw new Error('HyperCubeLayout is not valid');
+      if (dataPage.qArea && dataPage.qArea.qTop > qHeight) {
+        throw new Error('qDataPages are missing pages.');
+      }
+      qHeight += dataPage.qArea.qHeight;
+    }, this);
```
Comment (StefanEnberg) [inline src/hyper-cube.js]: We should evaluate if dataPage.qArea is defined and throw exception if not (Will crash on line 99 anyway but with a less expressive message). Then the dataPage.qArea check can be removed.

**Comment by StefanEnberg** [comment]
The task `test:coverage` is the on you one that is interesting when developing the other task is for uploading coverage to coveralls. Sorry for the confusion regarding coverage (I will make it clearer). CircleCi and coveralls shouldn't require any additional for contributors. Again sorry for the confusion

**Comment by KFredberg** [comment]
I like codeclimate, forces you to write cleaner code

**Comment by StefanEnberg** [comment]
Codeclimate was activated a few days ago :). I like it a lot. It summarized what I tried to convey in my comments but in a short and effective way. Thanks for your contribution (I will ping you regarding t-shirt size :) )

**Pull request merged**